test/common.sh

```diff
@@ -709,4 +709,44 @@ function create_delete_tests() {
       "SELECT deleted,created_at FROM account WHERE addr=decode('B9rLS22e0UGxdXa9RZrmQh1IbaPU7yJHxAmjlrguoiE=','base64') OR addr=decode('//////////////////////////////////////////8=','base64')" \
       "f|0
 f|0"
+
+      rest_test "[rest] b64 transaction fields are serialized" \
+      "/v2/transactions/TV5RPJFA6YT2APADUOYKIEL3NFAXSVB5J4JO6TSG7BHK4Z5OJKSA?pretty" \
+      200 \
+      true \
+      '"decimals": 0' \
+      '"default-frozen": false' \
+      '"name": "bogocoin"' \
+      '"name-b64": "Ym9nb2NvaW4="' \
+      '"reserve": "EQJSQTOITX64CPGA5VRKBKLEJR57YXVNLTW5DKZMESIPTEOLRDNWQIJCGU"' \
+      '"total": 1000000000000' \
+      '"unit-name": "bogo"' \
+      '"unit-name-b64": "Ym9nbw=="' \
+      '"created-asset-index": 36'
+
+      rest_test "[rest] b64 asset fields are serialized" \
+      "/v2/assets/36?pretty" \
+      200 \
+      true \
+      '"decimals": 0' \
+      '"default-frozen": false' \
+      '"name": "bogocoin"' \
+      '"name-b64": "Ym9nb2NvaW4="' \
+      '"reserve": "EQJSQTOITX64CPGA5VRKBKLEJR57YXVNLTW5DKZMESIPTEOLRDNWQIJCGU"' \
+      '"total": 1000000000000' \
+      '"unit-name": "bogo"' \
+      '"unit-name-b64": "Ym9nbw=="'
+
+      rest_test "[rest] b64 account fields are serialized" \
+      "/v2/accounts/EQJSQTOITX64CPGA5VRKBKLEJR57YXVNLTW5DKZMESIPTEOLRDNWQIJCGU?pretty" \
+      200 \
+      true \
+      '"decimals": 0' \
+      '"default-frozen": false' \
+      '"name": "bogocoin"' \
+      '"name-b64": "Ym9nb2NvaW4="' \
+      '"reserve": "EQJSQTOITX64CPGA5VRKBKLEJR57YXVNLTW5DKZMESIPTEOLRDNWQIJCGU"' \
+      '"total": 1000000000000' \
+      '"unit-name": "bogo"' \
+      '"unit-name-b64": "Ym9nbw=="'
```

Comment:
I think we also need tests where the strings are not utf-8-printable. I think a go test will suffice.

Reply:
Unfortunately, I don't have a good way to test the HTTP layer. You're right that this is missing the final check for removing name/unit-name/url if they contain non-utf8 characters.

Reply:
Mock `IndexerDb` and test the api functions? Ideally, we would test api functions with an implementation (implementations) of `IndexerDb`, but a bit of refactoring will need to be done beforehand.

Reply:
For now, I have run some tests manually.

I think it would be good to do some refactoring to the `api.Serve`, the pattern used by the block generator is pretty nice and allowed me to start/stop the server multiple times for the performance test runner.